overlord/snapstate/snapstate.go

```diff
@@ -307,17 +307,21 @@ func Install(st *state.State, name, channel string, revision snap.Revision, user
 		return nil, &snap.AlreadyInstalledError{Snap: name}
 	}
 
-	snapInfo, err := snapInfo(st, name, channel, revision, userID, flags)
+	info, err := snapInfo(st, name, channel, revision, userID)
 	if err != nil {
 		return nil, err
 	}
 
+	if !validInfoForFlags(info, &snapst, flags) {
+		return nil, fmt.Errorf("cannot find snap %q", name)
```

Comment:
yes. This particular error preserves (more or less) the current behaviour until then.